encoder/src/test/java/openlr/encoder/postprocessor/AlternatePathLrpProcessorTest.java

```diff
@@ -0,0 +1,46 @@
+package openlr.encoder.postprocessor;
+
+import openlr.OpenLRProcessingException;
+import openlr.PhysicalEncoder;
+import openlr.encoder.data.LocRefPoint;
+import openlr.encoder.database.TestMapStubTest;
+import openlr.encoder.properties.OpenLREncoderProperties;
+import openlr.map.teststubs.OpenLRMapDatabaseAdaptor;
+import openlr.properties.OpenLRPropertiesReader;
+import org.apache.commons.configuration.Configuration;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+public class AlternatePathLrpProcessorTest {
+    private OpenLRMapDatabaseAdaptor map;
+
+    @BeforeTest
+    public void loadMapStub(){
+        InputStream mapFile = OpenLRMapDatabaseAdaptor.class.getClassLoader().getResourceAsStream("teststubs/TestMapStub.xml");
+        this.map = OpenLRMapDatabaseAdaptor.from(mapFile);
+    }
+
+    @Test
+    public void verifyAlternatePathHandlerProcess() throws OpenLRProcessingException
+    {
+        Configuration encoderConfig = OpenLRPropertiesReader.loadPropertiesFromFile(new File(TestMapStubTest.class.getClassLoader().getResource("OpenLR-Encoder-Properties.xml").getFile()));
+        encoderConfig.setProperty("LrpAlternativePath",true);
+        encoderConfig.setProperty("AlternatePathRelativeThreshold",.20);
+        OpenLREncoderProperties properties = new OpenLREncoderProperties(encoderConfig, new ArrayList< PhysicalEncoder >());
```

Comment:
You don't need the explicit type `PhysicalEncoder` you can leave empty.

Reply:
d14a5b816e9b91dd99d4db70246353f967710764